.github/workflows/run-e2e-tests.yml

```diff
@@ -53,11 +53,11 @@ jobs:
       - name: Setup NodeJS
         uses: actions/setup-node@64ed1c7eab4cce3362f8c340dee64e5eaeef8f7c # v3.6.0
         with:
-          node-version: ${{ matrix.version }}
+          node-version: '20'
       - name: Setup dependencies
         uses: ./.github/actions/cached-node-modules
         with:
-          nodeVersion: ${{ matrix.version }}
+          nodeVersion: '20'
```

Comment:
Should this be `${{ matrix.version }}`?

Reply:
In an earlier version of this PR I left this as it was and integration tests were failing on Node.js 14 (see https://github.com/aws-powertools/powertools-lambda-typescript/actions/runs/6856373318).

This was because of CDK, which I had to upgrade in order to add support for the new runtime.

In upgrading the dependency however, we bumped to a version that dropped support for Node.js (the node version in the CI, not the Lambda runtime).

In order for Powertools to support the new Lambda runtime we needed the integration tests to be passing. At the same time, we can't drop the Node.js 14 runtime for 2 more weeks, so we also need the integration tests for that one to be running.

After some consideration I found an alternative that would have allowed us to keep the older CDK version and reference the new Node.js 20 runtime. Instead of using `Runtime.NODEJS_20_X` which is introduced in the new version, we can instead treat this as a custom runtime and use `new Runtime('nodejs20.x', RuntimeFamily.NODEJS)`.

This would have allowed us to keep the older version of CDK, run on Node.js 14 in the CI for a couple more weeks, and still deploy a function using the new runtime since CloudFormation would have recognized the `nodejs20.x` runtime at deployment time.

This worked fine (see https://github.com/aws-powertools/powertools-lambda-typescript/actions/runs/6856924987) but after thinking about it for some more time I also realized that the version of Node.js used in the CI doesn't really matter for integration tests.

The primary goal of these tests is to assert that our code is compatible with a certain Lambda runtime. To this end, we use CDK to deploy some ephemeral infrastructure, use `esbuild` to bundle the function code, and finally use the AWS SDK to interact with the deployed resources.

With this in mind, the only Node.js version that matters in this test is the one used in the Lambda runtime, which is completely independent from the one used in the CI. Using one Node.js version or the other in the CI doesn't affect the CDK deployment, nor the bundle (settings are codified in the `tsconfig.json`), or the SDK calls.

Given the above, I consider it safe to use the latest Node.js version in the CI while still using the matrix value in the Lambda runtime. This change does not affect the unit test matrix, which does run the test within the CI and as such still needs to be using the target Node.js versions.

What do you think? Am I missing something?

Reply:
> With this in mind, the only Node.js version that matters in this test is the one used in the Lambda runtime, which is completely independent from the one used in the CI. Using one Node.js version or the other in the CI doesn't affect the CDK deployment, nor the bundle (settings are codified in the tsconfig.json), or the SDK calls.

Makes sense, good change!